### ISSUE TITLE
Quadratic behavior when scanning inline HTML comments

### ISSUE THREAD

**Issue by nwellnhof** [commonmark/cmark#457]
```
python3 -c 'print("a"+"<!--"*50000)' |build/src/cmark >/dev/null
```

This regressed in commit 4470ff33. Found by OSS-Fuzz.

**Comment by jgm** [comment]
I already have a fix for this which I'll push later.

**Comment by jgm** [comment]
Fixed by 6a5126a8d65ad8615ed1f8efb025bebc06e9483d

**Comment by kevinbackhouse** [comment]
FYI, this bug also affected [cmark-gfm](https://github.com/github/cmark-gfm) (GitHub's fork of cmark), where we have assigned it [CVE-2023-22484](https://github.com/github/cmark-gfm/security/advisories/GHSA-24f7-9frr-5h2r).
